functions/generateZip.js

```diff
@@ -0,0 +1,105 @@
+import { queryFeatures } from '@esri/arcgis-rest-feature-service';
+import { getStorage } from 'firebase-admin/storage';
+import {
+  mkdirSync,
+  readFileSync,
+  readdirSync,
+  rmSync,
+  rmdirSync,
+  writeFileSync,
+} from 'fs';
+import gdal from 'gdal-async';
+import JSZip from 'jszip';
+import { downloadFormats } from './common/config.js';
+
+async function downloadLayer(layer, folder) {
+  console.log(`querying ${layer.name}...`);
+  // execute query against feature service
+  const geojson = await queryFeatures({
+    url: layer.featureService,
+    objectIds: layer.objectIds,
+    f: 'geojson',
+    returnGeometry: true,
+    outFields: '*',
+  });
+
+  // write geojson to temp file
+  const filePath = `${folder}/${layer.name
+    .split(' ')
+    .join('')}_${Date.now()}.geojson`;
+  writeFileSync(filePath, JSON.stringify(geojson));
+
+  return filePath;
+}
+
+function convert(geojsonPath, format) {
+  console.log(`converting ${geojsonPath} to ${format}...`);
+  const dataset = gdal.open(geojsonPath);
+  let driver;
+  let fileExtension;
+  switch (format) {
+    case downloadFormats.shapefile:
+      driver = gdal.drivers.get('ESRI Shapefile');
```

Comment:
**question**: are the dependencies required for this available to functions?

Reply:
Removed the function entirely.